cmd/rancher-desktop-guestagent/main.go

```diff
@@ -98,59 +97,61 @@ func main() {
 		cancel()
 	}()
 
-	tcpTracker := tcplistener.NewListenerTracker()
+	if !*enableContainerd && !*enableDocker {
+		log.Fatal("requires either -docker or -containerd enabled.")
+	}
 
-	wslAddr, err := getWSLAddr(wslInfName)
-	if err != nil {
-		log.Fatalf("failure getting WSL IP addresses: %v", err)
+	if *enableContainerd && *enableDocker {
+		log.Fatal("requires either -docker or -containerd, not both.")
 	}
 
-	forwarder := forwarder.NewVtunnelForwarder(*vtunnelAddr)
-	portTracker := tracker.NewPortTracker(forwarder, wslAddr)
+	var portTracker tracker.Tracker
 
 	if *enablePrivilegedService {
-		if !*enableContainerd && !*enableDocker {
-			log.Fatal("-privilegedService mode requires either -docker or -containerd enabled.")
-		}
-
-		if *enableContainerd && *enableDocker {
-			log.Fatal("-privilegedService mode requires either -docker or -containerd, not both.")
-		}
-
 		if *vtunnelAddr == "" {
 			log.Fatal("-vtunnelAddr must be provided when docker is enabled.")
 		}
 
-		if *enableContainerd {
-			group.Go(func() error {
-				eventMonitor, err := containerd.NewEventMonitor(*containerdSock, portTracker, tcpTracker)
-				if err != nil {
-					return fmt.Errorf("error initializing containerd event monitor: %w", err)
-				}
-				if err := tryConnectAPI(ctx, containerdSocketFile, eventMonitor.IsServing); err != nil {
-					return err
-				}
-				eventMonitor.MonitorPorts(ctx)
-
-				return eventMonitor.Close()
-			})
+		wslAddr, err := getWSLAddr(wslInfName)
+		if err != nil {
+			log.Fatalf("failure getting WSL IP addresses: %v", err)
 		}
 
-		if *enableDocker {
-			group.Go(func() error {
-				eventMonitor, err := docker.NewEventMonitor(portTracker)
-				if err != nil {
-					return fmt.Errorf("error initializing docker event monitor: %w", err)
-				}
-				if err := tryConnectAPI(ctx, dockerSocketFile, eventMonitor.Info); err != nil {
-					return err
-				}
-				eventMonitor.MonitorPorts(ctx)
-				eventMonitor.Flush()
-
-				return nil
-			})
-		}
+		forwarder := forwarder.NewVtunnelForwarder(*vtunnelAddr)
+		portTracker = tracker.NewPortTracker(forwarder, wslAddr)
+	} else {
+		portTracker = tracker.NewAPITracker(tracker.GatewayBaseURL)
```

Comment:
The `portTracker` isn't a `PortTracker`. Should that just be called `tracker` now?

Or maybe rename `PortTracker` to `VTunnelTracker`, and make the interface be `PortTracker`?

Reply:
The `tracker` can't be an option since it will conflict with the namespace, however, both `NewPortTracker` and `NewAPITracker` are still two different variations of port tracking mechanisms, they are both used to keep track of the ports (more specifically port mappings).

Reply:
This has been resolved — it's now `VTunnelTracker`.